packages/perseus-core/src/widgets/core-widget-registry.ts

```diff
@@ -83,6 +86,37 @@ export function isAccessible(
         ? accessible(widgetOptions)
         : !!accessible;
 }
+
+export const traverseChildWidgets = (
+    widgetInfo: PerseusWidget,
+    traverseRenderer: any,
+): PerseusWidget => {
+    if (!traverseRenderer) {
+        throw new PerseusError(
+            "traverseRenderer must be provided, but was not",
+            Errors.Internal,
+        );
+    }
+
+    // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
+    if (!widgetInfo || !widgetInfo.type || !widgets[widgetInfo.type]) {
+        return widgetInfo;
+    }
+
+    const widgetExports = widgets[widgetInfo.type];
+    const props = widgetInfo.options;
+
+    // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
+    if (widgetExports.traverseChildWidgets && props) {
+        const newProps = widgetExports.traverseChildWidgets(
+            props,
+            traverseRenderer,
+        );
+        return _.extend({}, widgetInfo, {options: newProps});
```

Comment:
We're working to remove usages of `underscore` (`_`) in our codebase.  In this case, this `extend()` function can be written in standard ES5 like this: 

```suggestion
        return {...widgetInfo, options: newProps};
```

Reply:
Thank you for the suggestion! I've replaced _.extend with object spread syntax for consistency with modern JavaScript and to support the effort to remove underscore usage.